### ISSUE TITLE
xlim (and ylim) break my plots

### ISSUE THREAD

**Issue by joshmaxsilverman** [yhat/ggplot#421]
I find that `xlim()` (and `ylim()`) break my plots (though only inline in iPython notebooks). For instance, calling the following

```
ggplot(aes(x='time', y='OD555', color='conc_GLY'), data=tpm_lim) +\
    geom_point() +\
    geom_line() +\
    xlim(0.115 , 0.140)
```

yields

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-266-64369a019002> in <module>()
----> 1 ggplot(aes(x='time', y='OD555', color='conc_GLY'), data=tpm_lim) +    geom_point() +    geom_line() +    ylab('OD$_{555}$') +    xlab('time (min)') +    xlim(0.115 , 0.140)

TypeError: unsupported operand type(s) for +: 'ggplot' and 'tuple'
```

-------

although this seems to be the same as the usage in a docs example (which also breaks for me):

```
ggplot(diamonds, aes(x='carat', y='price')) +\
    geom_point() +\
    ylim(2,3)
```

-------

I'm running ggplot (0.6.5), and matplotlib (1.4.3).

**Comment by has2k1** [comment]
@joshmaxsilverman , you are mixing ggplot and matplotlib interfaces. The correct ggplot way 

```python
ggplot(aes(x='time', y='OD555', color='conc_GLY'), data=tpm_lim) +\
    geom_point() +\
    geom_line() +\
    scale_x_continuous(limits=[0.115 , 0.140])
```

**Comment by joshmaxsilverman** [comment]
Thanks! All fixed now.

I'm still a bit confused, it seems in the docs example, `xlim` is used as part of the `ggplot` interface: http://ggplot.yhathq.com/docs/xlim.html

Do you know why this is?

**Comment by has2k1** [comment]
They are upkeep issues, some parts of the code base changed and the unit-tests didn't pick it up. It should clear up in the near future.

**Comment by glamp** [comment]
fixed in 0.9.3
